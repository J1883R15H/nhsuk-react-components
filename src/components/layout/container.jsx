import React from 'react';
import PropTypes from 'prop-types';
import stylePropType from 'react-style-proptype';
import classNames from 'classnames';

const Container = ({ fluid, children, className, style }) => (
  <div
    className={classNames(
      { 'nhsuk-width-container': !fluid },
      { 'nhsuk-width-container-fluid': fluid },
      className
    )}
    style={style}
  >
    {children}
  </div>
);

Container.propTypes = {
  fluid: PropTypes.bool,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  style: stylePropType
};

Container.defaultProps = {
  fluid: false,
  className: '',
  style: {}
};

export default Container;
