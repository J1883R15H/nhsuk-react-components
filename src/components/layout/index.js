import Container from './container.jsx';
import Row from './row.jsx';

export { Container, Row };
