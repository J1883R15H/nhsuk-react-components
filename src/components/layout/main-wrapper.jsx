import React from 'react';
import { node, string } from 'prop-types';
import stylePropType from 'react-style-proptype';
import classNames from 'classnames';

const MainWrapper = ({ children, id, className, style }) => (
  <main
    className={classNames('nhsuk-main-wrapper', className)}
    style={style}
    id={id}
  >
    {children}
  </main>
);

MainWrapper.propTypes = {
  children: node.isRequired,
  id: string,
  style: stylePropType,
  className: string
};

MainWrapper.defaultProps = {
  id: '',
  style: {},
  className: ''
};

export default MainWrapper;
