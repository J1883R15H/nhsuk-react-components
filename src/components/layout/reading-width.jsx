import React from 'react';
import { node, string } from 'prop-types';
import stylePropType from 'react-style-proptype';
import classNames from 'classnames';

const ReadingWidth = ({ children, className, style }) => (
  <div className={classNames('nhsuk-u-reading-width', className)} style={style}>
    {children}
  </div>
);

ReadingWidth.propTypes = {
  children: node.isRequired,
  className: string,
  style: stylePropType
};

ReadingWidth.defaultProps = {
  className: '',
  style: {}
};

export default ReadingWidth;
