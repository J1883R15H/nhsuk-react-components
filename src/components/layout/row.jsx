import React from 'react';
import PropTypes from 'prop-types';
import stylePropType from 'react-style-proptype';
import classNames from 'classnames';

const Row = ({ children, className, style }) => (
  <div className={classNames('nhsuk-grid-row', className)} style={style}>
    {children}
  </div>
);
const Column = ({
  children,
  width,
  tabletWidth,
  forceLayout,
  className,
  style
}) => (
  <div
    className={classNames(
      `nhsuk-grid-column-${width}`,
      { [`nhsuk-u-${width}`]: forceLayout },
      { [`nhsuk-u-${tabletWidth}-tablet`]: tabletWidth },
      className
    )}
    style={style}
  >
    {children}
  </div>
);

Row.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  style: stylePropType
};

Row.defaultProps = {
  children: null,
  className: '',
  style: {}
};

Column.propTypes = {
  width: PropTypes.oneOf([
    'full',
    'three-quarters',
    'one-half',
    'two-thirds',
    'one-third',
    'one-quarter'
  ]).isRequired,
  children: PropTypes.node,
  forceLayout: PropTypes.bool,
  tabletWidth: PropTypes.oneOf([
    '',
    'full',
    'three-quarters',
    'one-half',
    'two-thirds',
    'one-third',
    'one-quarter'
  ]),
  className: PropTypes.string,
  style: stylePropType
};

Column.defaultProps = {
  children: null,
  forceLayout: false,
  tabletWidth: '',
  className: '',
  style: {}
};

Row.Column = Column;

export default Row;
