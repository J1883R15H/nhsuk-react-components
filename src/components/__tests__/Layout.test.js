import React from 'react';
import { shallow } from 'enzyme';
import { Container, MainWrapper, ReadingWidth, Row } from '../layout';

describe('Layout', () => {
  describe('Container', () => {
    it('matches snapshot', () => {
      const shallowMount = shallow(<Container>Child</Container>);
      expect(shallowMount).toMatchSnapshot();
      shallowMount.unmount();
    });
    it('renders a fluid container', () => {
      const standard = shallow(<Container>Child</Container>);
      const fluid = shallow(<Container fluid>Child</Container>);
      expect(
        standard.containsMatchingElement(
          <div className="nhsuk-width-container">Child</div>
        )
      ).toBeTruthy();
      expect(
        fluid.containsMatchingElement(
          <div className="nhsuk-width-container-fluid">Child</div>
        )
      ).toBeTruthy();
      standard.unmount();
      fluid.unmount();
    });
  });
  describe('MainWrapper', () => {
    it('matches snapshot', () => {
      const shallowMount = shallow(<MainWrapper>Child</MainWrapper>);
      expect(shallowMount).toMatchSnapshot();
      expect(
        shallowMount.containsMatchingElement(
          <main className="nhsuk-main-wrapper">Child</main>
        )
      ).toBeTruthy();
      shallowMount.unmount();
    });
  });
  describe('ReadingWidth', () => {
    it('matches snapshot and element', () => {
      const shallowMount = shallow(<ReadingWidth>Text</ReadingWidth>);
      expect(
        shallowMount.containsMatchingElement(
          <div className="nhsuk-u-reading-width">Text</div>
        )
      ).toBeTruthy();
      expect(shallowMount).toMatchSnapshot();
      shallowMount.unmount();
    });
  });
  describe('Row', () => {
    it('matches snapshot and element', () => {
      const shallowMount = shallow(<Row>Child</Row>);
      expect(shallowMount).toMatchSnapshot();
      expect(
        shallowMount.containsMatchingElement(
          <div className="nhsuk-grid-row">Child</div>
        )
      ).toBeTruthy();
      shallowMount.unmount();
    });
  });
  describe('Column', () => {
    it('matches snapshot and element', () => {
      const shallowMount = shallow(<Row.Column width="full">Child</Row.Column>);
      expect(shallowMount).toMatchSnapshot();
      expect(
        shallowMount.containsMatchingElement(
          <div className="nhsuk-grid-column-full">Child</div>
        )
      ).toBeTruthy();
      shallowMount.unmount();
    });
    it('forces layout', () => {
      const shallowMount = shallow(
        <Row.Column width="full" forceLayout>
          Child
        </Row.Column>
      );
      expect(shallowMount).toMatchSnapshot();
      expect(
        shallowMount.containsMatchingElement(
          <div className="nhsuk-grid-column-full nhsuk-u-full">Child</div>
        )
      ).toBeTruthy();
      shallowMount.unmount();
    });
    it('works with tablet width', () => {
      const shallowMount = shallow(
        <Row.Column width="full" tabletWidth="one-half">
          Child
        </Row.Column>
      );
      expect(shallowMount).toMatchSnapshot();
      expect(
        shallowMount.containsMatchingElement(
          <div className="nhsuk-grid-column-full nhsuk-u-one-half-tablet">
            Child
          </div>
        )
      ).toBeTruthy();
      shallowMount.unmount();
    });
  });
});
