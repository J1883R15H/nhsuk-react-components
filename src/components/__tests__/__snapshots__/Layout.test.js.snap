// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Column forces layout 1`] = `
<div
  className="nhsuk-grid-column-full nhsuk-u-full"
  style={Object {}}
>
  Child
</div>
`;

exports[`Layout Column matches snapshot and element 1`] = `
<div
  className="nhsuk-grid-column-full"
  style={Object {}}
>
  Child
</div>
`;

exports[`Layout Column works with tablet width 1`] = `
<div
  className="nhsuk-grid-column-full nhsuk-u-one-half-tablet"
  style={Object {}}
>
  Child
</div>
`;

exports[`Layout Container matches snapshot 1`] = `
<div
  className="nhsuk-width-container"
  style={Object {}}
>
  Child
</div>
`;

exports[`Layout MainWrapper matches snapshot 1`] = `
<main
  className="nhsuk-main-wrapper"
  id=""
  style={Object {}}
>
  Child
</main>
`;

exports[`Layout ReadingWidth matches snapshot and element 1`] = `
<div
  className="nhsuk-u-reading-width"
  style={Object {}}
>
  Text
</div>
`;

exports[`Layout Row matches snapshot and element 1`] = `
<div
  className="nhsuk-grid-row"
  style={Object {}}
>
  Child
</div>
`;
